### DISCUSSION TITLE
Variable for use Only in Boundary Condition

### DISCUSSION THREAD

**Discussion by AngeloCass** [idaholab/moose#18141] [Q&A General] [answered]
Hello,

I am wondering if there is a way to have a variable that exists only in a boundary condition and does not require use in a kernel. This variable only exists at the surface or 'top' of the domain or 'mesh'. The variable is coupled with two others in the governing equation. Optimally, I would like to solve for this value only on the surface and neglect it elsewhere. I have not found an effective way to do this. If anyone could provide information on how I may accomplish this, it would be much appreciated.

Thanks!

**Comment by hugary1995** [comment, marked as answer]
You can define a block for your sideset/nodeset of interest, then claim the variable to be block-restricted, i.e.
```
[Variables]
  [u]
    block = top
  []
[]
```
You may or may not need to turn off the kernel coverage check depending on your setup.

**Reply by WilkAndy** [reply]
Perhaps `LowerDBlockFromSidesetGenerator` or `ParsedSubdomainMeshGenerator` will be of use.  @hugary1995 is suggesting you'll have to include the following in your input file

```
[Problem]
  kernel_coverage_check = false
[]
```
